### PR TITLE
feat: added ability to create Single Production plan from multiple Sales orders

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
Ref: [Asana](https://app.asana.com/0/1192407896547052/1199693860932062)

pp-from-so

### CODE REVIEW

File: erpnext/selling/doctype/sales_order/sales_order_list.js [hunk 1]
```diff
@@ -351,8 +351,56 @@ frappe.listview_settings['Sales Order'] = {
 			}
 		};
 
+		const create_one_production_plan_action = () => {
+			const selected_docs = listview.get_checked_items();
+			const docnames = listview.get_checked_items(true);
+
+			if (selected_docs.length > 0) {
```
Comment (hrwX) [inline erpnext/selling/doctype/sales_order/sales_order_list.js:358]:
```suggestion
			if (selected_docs.length === 0) {
```
- Reduce the indentation

**Review by hrwX — APPROVED** [review]

File: erpnext/selling/doctype/sales_order/sales_order_list.js (hunk 1 above)
Comment (MyuddinKhatri) [reply erpnext/selling/doctype/sales_order/sales_order_list.js:358]: > * Reduce the indentation

Whole execution is performed inside this block and the error thrown after this is for  for **On Hold** and **Closed** Orders.

Comment (hrwX) [reply erpnext/selling/doctype/sales_order/sales_order_list.js:358]: If the length is equal to zero, do not perform it, just wanted you to reduce the indentation
```suggestion
			if (selected_docs.length === 0) { return; }
```

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/selling/doctype/sales_order/sales_order_list.js (hunk 1 above)
Comment (sahil28297) [reply erpnext/selling/doctype/sales_order/sales_order_list.js:358]: @hrwX I think the current indentation seems more intuitive.

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**